R/join_ald_scenario.R

```diff
@@ -42,14 +42,27 @@ join_ald_scenario <- function(data,
   region_isos <- region_isos %>%
     dplyr::rename(scenario_source = .data$source)
 
-  data %>%
+  out <- data %>%
     left_join(ald, by = ald_columns()) %>%
+    warn_if_has_zero_rows() %>%
     dplyr::inner_join(scenario, by = scenario_columns()) %>%
+    warn_if_has_zero_rows() %>%
     dplyr::mutate(plant_location = tolower(.data$plant_location)) %>%
     dplyr::inner_join(
       region_isos,
       by = c("region", "plant_location" = "isos", "scenario_source")
-    )
+    ) %>%
+    warn_if_has_zero_rows()
+
+  out
+}
+
+warn_if_has_zero_rows <- function(data) {
+  if (identical(nrow(data), 0L)) {
+    rlang::warn(class = "has_zero_row", "`data` has zero-rows.")
+  }
+
+  invisible(data)
```

Comment:
I'm considering adding the new argument `message` to `warn_if_has_zero_rows()` so that we can throw different warnings at each step in the pipeline. 
Not yet sure -- might be an overkill.